config/v1/types_cluster_version.go

```diff
@@ -408,6 +422,15 @@ type URL string
 // Update represents an administrator update request.
 // +k8s:deepcopy-gen=true
 type Update struct {
+	// architecture is an optional field that indicates the desired
+	// value of the cluster architecture. In this context cluster
+	// architecture means either homogeneous or heterogeneous.
+	// Architecture is ignored if image is specified. The empty
+	// value indicates that the field is not set.
```

Comment:
What happens if it is not set?

Reply:
if `image` is set, both `version` and `architecture` are ignored.  Which is covered here with "Architecture is ignored if image is specified".

If `image` is not set, `version` and `architecture` are still both optional when looking up targets in `availableUpdates`, and both default to "the release that I'm currently reconciling".  Maybe we can cover that with wording in `desiredUpdate` like:

> desiredUpdate is an optional field that indicates the desired value of the cluster version. Setting this value will request an update if the requested target does not match the current target release.  The set of recommended update values is listed as part of available updates in status, and setting values outside that range may cause the upgrade to fail.  You may specify an image pullspec to request updates regardless of the recommended available updates.  In the absence of an image pullspec, you may set either or both of version and architecture to manage lookup in available updates, with both properties defaulting to the value for the current target release.

or some such...

Reply:
Right. I generally followed the pattern of `version` but can add something explicit.

Reply:
why should we allow an ignored field to be set instead of rejecting ignored and potentially conflicting data and forcing the caller to write a guaranteed non-conflicting manifest?

Reply:
> What happens if it is not set?
The short answer is we use the current value of architecture which CVO knows. 

> why should we allow an ignored field to be set instead of rejecting ignored
Assuming this is, if `architecture` is set why should we allow `version` to be set? If so, I'm not sure we do. When they make the first update to heter they'll set arch to `multi` and version should not be set because they can only upgrade to current version. After that there's no reason to ever set `arch` again because it can only be `multi' or "not set" so everything they need to do can be done by just setting version (or image).